converter.go

```diff
@@ -2162,6 +2164,38 @@ func arrowToRecordSingleColumn(ctx context.Context, field arrow.Field, col arrow
 	return newCol, nil
 }
 
+// returns a pointer to a new array which will be populated if we converted the array to valid utf8
+// or returns null, and an indicator that we should retain the original column
+func arrowStringRecordToColumn(
+	ctx context.Context,
+	stringCol *array.String,
+	mem memory.Allocator,
+	numRows int64,
+	fieldMetadata fieldMetadata,
+) arrow.Array {
+	if arrowBatchesUtf8ValidationEnabled(ctx) && stringCol.DataType().ID() == arrow.STRING {
+		tb := array.NewStringBuilder(mem)
+		defer tb.Release()
+
+		for i := 0; i < int(numRows); i++ {
+			if stringCol.IsValid(i) {
+				stringValue := stringCol.Value(i)
+				if !utf8.ValidString(stringValue) {
+					logger.WithContext(ctx).Error("Invalid UTF-8 characters detected while reading query response, column: ", fieldMetadata.Name)
+					stringValue = strings.ToValidUTF8(stringValue, "�")
+				}
+				tb.Append(stringValue)
+			} else {
+				tb.AppendNull()
+			}
+		}
+		arr := tb.NewArray()
+		return arr
+	}
+	stringCol.Retain()
+	return nil
```

Comment:
If we return `stringCol` here instead of `nil` we can omit nil checking when this method is called, what do you think?

Reply:
Done!